### ISSUE TITLE
Can't download dataset!

### ISSUE THREAD

**Issue by njugh** [facebookresearch/DONERF#5]
Great work! I'm very interested in your work and want to test with your datasets. Unfortunately, I can't download your data at https://repository.tugraz.at/records/jjs3x-4f133. Could you please share the data on GoogleDrive? Thank you for your help!

**Comment by thomasneff** [comment]
Hi!

Can you tell me how it doesn't work and what exactly your issue is when trying to download?

If you scroll down, you should see the individual zip files for each scene, and you should be able to download them. Maybe our server had some issues at some point?

![image](https://user-images.githubusercontent.com/3900708/132515610-6cf721f3-3232-4766-95ae-f6b2f54d613e.png)

**Comment by xmk2222** [comment]
Same Issue here

**Comment by xmk2222** [comment]
Could be downloaded using wget

**Comment by Realitian** [comment]
Hi @yyeboah, @zhaoyang-lv 
Thank you for works.
Can you explain how to make own dataset?

**Comment by thomasneff** [comment]
> Hi @yyeboah, @zhaoyang-lv
> Thank you for works.
> Can you explain how to make own dataset?

Hi!

We generated the datasets using Blender, by using python scripts that hook into the images, camera and depth maps that are rendered by Blender.

If you wanted to create your own dataset, I recommend looking at one of the `transforms_train.json` files inside one of the scenes in our dataset. 

Essentially, you need the following attributes in each .json file, which we adapted from the format also used by [NeRF](https://github.com/bmild/nerf)

`camera_angle_x`: Horizontal FOV of the camera.

`view cell center`: This is the center of the view cell that will be used to generate images from. For our definition of the view cell, I recommend reading the relevant section of the paper, but it's essentially just a box that limits allowed range of the camera positions, and we use this for normalization in our code.

`view cell size`: This is the size of the view cell, and is also used for normalization in the code.

`frames: [...]`: This is an array of objects that refer to the individual examples, and each object contains:
   `file_path`: the relative path to the image/depth map
   `transform_matrix`: this is the camera matrix (exported as a list) that is used to transform the rays when training, and directly corresponds to the `matrix_world` of the Blender camera object, if you're exporting from blender.

This is essentially everything you need within the json files - apart from that you simply need to put the images and depth maps in the correct file paths (as specified in the json files). The images are standard .pngs, and the depth is exported as raw numpy binary arrays that are reshaped accordingly in our input processing code. For the exported depth, you need to ensure that this encodes the correct distance for each ray, and is not a fixed distance from the camera origin, otherwise the depth oracles will struggle to learn correctly.

And this should be all there is to it - if you have your own data, I recommend looking at the existing datasets and the input loading code in `src/datasets.py`, which should be sufficient to adapt your own data accordingly.

**Comment by Kim9878** [comment]
Hi @thomasneff
I'm now following your instruction to create my own dataset, but there are some attributes I don't know how to set.
Could you tell me how to decide a suitable view cell size and view cell center? And I also want to check which coordinate system the transform matrix is?
Thank you for your help!

**Comment by thomasneff** [comment]
Hi!

About the view cells: You can pretty much choose that depending on your scene and size of the region you will later want to generate new views from.

For our scenes, we chose them such that they don't contain any objects within the view cell, and if you generate enough training data, you can generate pretty large view cells (we've tested view cells that were 3-5x as large as the ones we used in the paper, and it still worked fine.)

We use Blender's coordinate system, as we directly exported the matrices from there.

**Comment by chensjtu** [comment]
Hi! I wonder if you can share the blender rendering code? Since I'm not familiar with blender, I may render additional images in the scene classroom with a detailed rendering script.

**Comment by kwea123** [comment]
@thomasneff I've tried downloading using the link or `wget`, but the download always stops halfway.
![Screenshot from 2022-08-03 10-16-44](https://user-images.githubusercontent.com/11364490/182503323-c56fd8ff-989d-44f2-95b6-f5ad21cbe916.png)

This is what I get from `wget`, it keeps being interrupted and restarted, but the download never finishes. Could you please verify the download link, or upload to a better accessible platform like google drive? Thank you. I would like to reproduce your results but gets stumped at downloading

**Comment by chensjtu** [comment]
> @thomasneff I've tried downloading using the link or `wget`, but the download always stops halfway. ![Screenshot from 2022-08-03 10-16-44](https://user-images.githubusercontent.com/11364490/182503323-c56fd8ff-989d-44f2-95b6-f5ad21cbe916.png)
> 
> This is what I get from `wget`, it keeps being interrupted and restarted, but the download never finishes. Could you please verify the download link, or upload to a better accessible platform like google drive? Thank you. I would like to reproduce your results but gets stumped at downloading

Same Issue, but I got the classroom and barbershop. I can share them with you if you need.

**Comment by chensjtu** [comment]
I've just uploaded what I have downloaded. @kwea123  
https://drive.google.com/drive/folders/1f-pKizJoa5fB3JA7zQdlrEC_lHII50uy?usp=sharing

**Comment by joeyw1030** [comment]
您好为什么我用blender生成的深度是扭曲的，就是从原点到远处呈现一个个圈。
